src/App.tsx

```diff
@@ -15,71 +16,84 @@ export const goods = [
-            </button>
-          </td>
+  removeGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+    return (
+      <main className="section container">
+        {selectedGood !== ''
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              { selectedGood }
+              {' '}
+              is selected
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.removeGood}
+              />
+            </h1>
+          )
+          : <h1 className="title">No goods selected</h1>}
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const hasGood = selectedGood === good;
+              return (
+                <tr
+                  className={classNames('', {
+                    'has-background-success-light': hasGood,
+                  })}
+                  key={good}
+                  data-cy="Good"
+                >
+                  <td>
+                    <button
+                      data-cy={classNames('', {
+                        RemoveButton: hasGood,
+                        AddButton: !hasGood,
+                      })}
+                      type="button"
+                      className={classNames('button', {
+                        'is-info': hasGood,
+                      })}
+                      onClick={
+                        hasGood
+                          ? this.removeGood
+                          : () => this.addGood(good)
+                      }
+                    >
+                      {hasGood ? '-' : '+'}
+                    </button>
+                  </td>
+                  <td data-cy="GoodTitle" className="is-vcentered">
+                    {good}
+                  </td>
+                </tr>
+              );
+            })}
```

Comment:
if you change almost everything depends on `hasGood` I think it will be better to rewrite it like this
```
{hasGood ? (
  <tr> .... </tr>
) : (
   <tr> .... </tr>
)}
```